hide private builtins from exception stack traces

dcrousso opened the pull request:
#### 8d17ce89dc8babf5bf8383ed3f6402a693c21871
<pre>
hide private builtins from exception stack traces
<a href="https://bugs.webkit.org/show_bug.cgi?id=242980">https://bugs.webkit.org/show_bug.cgi?id=242980</a>

Reviewed by Yusuke Suzuki.

Private builtins are really just an implementation detail, and as such should not be exposed to developers.

* Source/JavaScriptCore/interpreter/Interpreter.cpp:
(JSC::GetStackTraceFunctor::operator() const):

* Source/JavaScriptCore/inspector/ScriptCallStackFactory.cpp:
(Inspector::CreateScriptCallStackFunctor::operator() const):
Drive-by: Move the `ImplementationVisibility` check after the `needToSkipAFrame` check to make it
          easier for callers to reason about the value to provide.

* LayoutTests/inspector/console/message-stack-trace-expected.txt:
* LayoutTests/inspector/debugger/js-stacktrace-expected.txt:
* LayoutTests/inspector/model/remote-object/error-expected.txt:

Canonical link: <a href="https://commits.webkit.org/252693@main">https://commits.webkit.org/252693@main</a>
</pre>

Constellation commented on Source/JavaScriptCore/interpreter/Interpreter.cpp:
```diff
@@ -392,6 +392,11 @@ class GetStackTraceFunctor {
+            if (codeBlock->ownerExecutable()->implementationVisibility() != ImplementationVisibility::Public)
+                return IterationStatus::Continue;
+        }
+
```
Move this after `m_framesToSkip`'s clause. This skip number is internal one, so including private one too makes coding it easier.

Constellation reviewed (approved):
r=me with comment.

dcrousso committed:
hide private builtins from exception stack traces
https://bugs.webkit.org/show_bug.cgi?id=242980

Reviewed by Yusuke Suzuki.

Private builtins are really just an implementation detail, and as such should not be exposed to developers.

* Source/JavaScriptCore/interpreter/Interpreter.cpp:
(JSC::GetStackTraceFunctor::operator() const):

* Source/JavaScriptCore/inspector/ScriptCallStackFactory.cpp:
(Inspector::CreateScriptCallStackFunctor::operator() const):
Drive-by: Move the `ImplementationVisibility` check after the `needToSkipAFrame` check to make it
          easier for callers to reason about the value to provide.

* LayoutTests/inspector/console/message-stack-trace-expected.txt:
* LayoutTests/inspector/debugger/js-stacktrace-expected.txt:
* LayoutTests/inspector/model/remote-object/error-expected.txt:

Canonical link: https://commits.webkit.org/252693@main
